From 92e0a2112b0f4e65d5770ebdf6e36ccb7a153b88 Mon Sep 17 00:00:00 2001
From: Stephen L Arnold <nerdboy@gentoo.org>
Date: Sun, 25 Apr 2021 21:05:22 -0700
Subject: [PATCH] add missing bits for mmc rescan patch (was part of bcmdhd
 patch)

Signed-off-by: Stephen L Arnold <nerdboy@gentoo.org>
---
 include/linux/mmc/host.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/linux/mmc/host.h b/include/linux/mmc/host.h
index c079b932330f..065358aad487 100644
--- a/include/linux/mmc/host.h
+++ b/include/linux/mmc/host.h
@@ -142,6 +142,8 @@ struct mmc_host_ops {
 	/* Check if the card is pulling dat[0:3] low */
 	int	(*card_busy)(struct mmc_host *host);
 
+	int     (*set_sdio_status)(struct mmc_host *host, int val);
+
 	/* The tuning command opcode value is different for SD and eMMC cards */
 	int	(*execute_tuning)(struct mmc_host *host, u32 opcode);
 
-- 
2.30.0

