From da2efd4cdf3dbe806f8005dcc641ab0f057d1e8f Mon Sep 17 00:00:00 2001
From: Stephen L Arnold <nerdboy@gentoo.org>
Date: Sun, 25 Apr 2021 14:27:29 -0700
Subject: [PATCH] arch: riscv: enable pm_sleep for beaglev bcmdhd testing

Signed-off-by: Stephen L Arnold <nerdboy@gentoo.org>
---
 arch/riscv/Kconfig | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/arch/riscv/Kconfig b/arch/riscv/Kconfig
index 44377fd7860e..eb66affa7876 100644
--- a/arch/riscv/Kconfig
+++ b/arch/riscv/Kconfig
@@ -423,6 +423,17 @@ menu "Power management options"
 
 source "kernel/power/Kconfig"
 
+config ARCH_HIBERNATION_POSSIBLE
+	def_bool y
+	depends on CPU_PM
+
+config ARCH_HIBERNATION_HEADER
+	def_bool y
+	depends on HIBERNATION
+
+config ARCH_SUSPEND_POSSIBLE
+	def_bool y
+
 endmenu
 
 source "drivers/firmware/Kconfig"
-- 
2.30.0

